classes/robot/crawler.php

```diff
@@ -549,53 +551,58 @@ public function process_queue($verbose = false) {
             $recentcourses = $this->get_recentcourses();
         }
 
-        // Iterate through the queue until we find an item that is a recent course, or the time runs out.
+        // Iterate through the queue.
```

Comment:
Isn't this nested while loop here redundant:

https://github.com/catalyst/moodle-tool_crawler/blob/master/lib.php#L102-L107